cvmfs/sync_mediator.cc

```diff
@@ -94,6 +95,33 @@ void SyncMediator::Add(SharedPtr<SyncItem> entry) {
     return;
   }
 
+  // .cvmfsbundles file type
+  if (entry->IsBundleSpec() && entry->IsRegularFile() &&
+      !entry->HasHardlinks()) {
+    std::string parent_path = GetParentPath(entry->GetUnionPath());
+    if (!IsMountPoint(parent_path)) {
```

Comment:
`IsMountPoint` probably works but is not the correct check.  We should rather check `parent_path != SyncUnion::union_path()`